R/multivariate_karcher_mean.R

```diff
@@ -1,124 +1,165 @@
+#' original input data.
+#' - `q`: A numeric array of shape \eqn{L \times M \times N} storing the SRVFs
+#' of the input data.
+#' - `betan`: A numeric array of shape \eqn{L \times M \times N} storing the
+#' aligned, possibly optimally rotated and optimally scaled, input data.
+#' - `qn`: A numeric array of shape \eqn{L \times M \times N} storing the SRVFs
+#' of the aligned, possibly optimally rotated and optimally scaled, input data.
+#' - `betamean`: A numeric array of shape \eqn{L \times M} storing the Karcher
+#' mean or median of the input data.
+#' - `qmean`: A numeric array of shape \eqn{L \times M} storing the Karcher mean
+#' or median of the SRVFs of the input data.
+#' - `type`: A character string indicating whether the Karcher mean or median
+#' has been returned.
+#' - `E`: A numeric vector storing the energy of the Karcher mean or median at
+#' each iteration.
+#' - `qun`: A numeric vector storing the cost function of the Karcher mean or
+#' median at each iteration.
+#'
 #' @keywords srvf alignment
+#'
 #' @references Srivastava, A., Klassen, E., Joshi, S., Jermyn, I., (2011). Shape
-#'    analysis of elastic curves in euclidean spaces. Pattern Analysis and Machine
-#'    Intelligence, IEEE Transactions on 33 (7), 1415-1428.
+#'   analysis of elastic curves in euclidean spaces. Pattern Analysis and
+#'   Machine Intelligence, IEEE Transactions on 33 (7), 1415-1428.
+#'
 #' @export
+#'
 #' @examples
 #' out <- multivariate_karcher_mean(beta[, , 1, 1:2], maxit = 2)
 #' # note: use more functions, small for speed
-multivariate_karcher_mean <- function (beta, lambda = 0.0, maxit = 20, ms = "mean")
+multivariate_karcher_mean <- function(beta,
+                                      rotation = FALSE,
+                                      scale = FALSE,
+                                      maxit = 20,
+                                      ms = c("mean", "median"))
 {
-  if(ms!="mean"&ms!="median"){warning("ms must be either \"mean\" or \"median\". ms has been set to \"mean\"",immediate. = T)}
-  if(ms!="median"){ms = "mean"}
+  ms <- rlang::arg_match(ms)
 
-  tmp = dim(beta)
-  n = tmp[1]
-  T1 = tmp[2]
-  time1 <- seq(0,1,length.out=T1)
-  N = tmp[3]
-  q = array(0, c(n, T1, N))
-  for (ii in 1:N) {
-    beta1 = beta[ , , ii]
-    out = curve_to_q(beta1, FALSE)
-    q[, , ii] = out$q
-  }
-
-  mu = q[, , 1]
-  bmu = beta[, , 1]
-  delta = 0.5
-  tolv = 1e-04
-  told = 5 * 0.001
-  itr = 1
-  sumd = rep(0, maxit + 1)
-  sumd[1] = Inf
-  betan = array(0, c(n, T1, N))
-  qn = array(0, c(n, T1, N))
-  normbar = rep(0, maxit + 1)
-  if(ms == "median"){ #run for median only, saves memory if getting mean
-    d_i = rep(0,N) #include vector for norm calculations
-    v_d = array(0, c(n, T1, N)) #include array to hold v_i / d_i
-  }
+  dims <- dim(beta)
+  L <- dims[1] # Dimension of codomain
+  M <- dims[2] # Size of the evaluation grid
+  N <- dims[3] # Sample size
+  q <- array(dim = c(L, M, N)) # Array for hosting SRVFs
 
```

Comment:
I do not want to use future in this package, users that use it do not use it and do not want to introduce another parallel framework unless the entire package is converted. So please remove

Reply:
Will do.